app/javascript/controllers/preview_controller.js

```diff
@@ -0,0 +1,17 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["inputPreText", "outputPreText", "inputPostText", "outputPostText"];
+
+  connect() {
+    this.updatePreTextOutput();
```

Comment:
This is also running on Builders#new form. We don't need to call the handlers when the prompt form is initially blank. I think it's better to exclude these calls for Builders#new page.

Reply:
Condition added if value is not present in input field, these functions will not be executed.